heudiconv/bids.py

```diff
@@ -487,3 +510,505 @@ def convert_sid_bids(subject_id):
     lgr.warning('{0} contained nonalphanumeric character(s), subject '
                 'ID was cleaned to be {1}'.format(subject_id, sid))
     return sid, subject_id
+
+
+def get_shim_setting(json_file):
+    """
+    Gets the "ShimSetting" field from a json_file.
+    If no "ShimSetting" present, return error
+
+    Parameters:
+    ----------
+    json_file : str
+
+    Returns:
+    -------
+    str with "ShimSetting" value
+    """
+    data = load_json(json_file)
+    try:
+        shims = data[SHIM_KEY]
+    except KeyError as e:
+        lgr.error('File %s does not have "ShimSetting".'
+                  'Please use a different "matching_parameters" in your heuristic file', json_file)
```

Comment:
```suggestion
        lgr.error('File %s does not have "%s". '
                  'Please use a different "matching_parameters" in your heuristic file',
                   json_file, SHIM_KEY)
```
to not hardcode what is stored in SHIM_KEY